Kernel/Syscalls/socket.cpp

```diff
@@ -199,32 +198,47 @@ ErrorOr<FlatPtr> Process::sys$sendmsg(int sockfd, Userspace<const struct msghdr*
             Thread::current()->send_signal(SIGPIPE, &Process::current());
         return EPIPE;
     }
-    auto data_buffer = TRY(UserOrKernelBuffer::for_user_buffer((u8*)iovs[0].iov_base, iovs[0].iov_len));
 
-    while (true) {
-        while (!description->can_write()) {
-            if (!description->is_blocking()) {
-                return EAGAIN;
+    Checked<ssize_t> total_length = 0;
+    for (auto const& iov : iovs)
+        total_length += iov.iov_len;
+
+    if (total_length.has_overflow())
+        return EINVAL;
+
+    size_t nwrite = 0;
+    for (auto const& iov : iovs) {
+        auto data_buffer = TRY(UserOrKernelBuffer::for_user_buffer((u8*)iov.iov_base, iov.iov_len));
+
+        while (true) {
+            while (!description->can_write()) {
+                if (!description->is_blocking()) {
+                    return EAGAIN;
+                }
```

Comment:
You can leave out the braces for single statement if-bodies - same for the block a couple of lines down.